crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -460,8 +460,8 @@ impl<'src> Parser<'src> {
     /// field will be [`ExprContext::Invalid`].
     ///
     /// See: <https://docs.python.org/3/reference/expressions.html#atom-identifiers>
-    pub(super) fn parse_name(&mut self) -> ast::ExprName {
-        let identifier = self.parse_identifier();
+    pub(super) fn parse_name(&mut self, recovery: IdentifierRecovery) -> ast::ExprName {
```

Comment:
I added this `recovery` context for better recovery when we see `@def` (parse it as a decorator followed by a function definition)